### PR TITLE
Adding debug logs for lookback feature

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Adding debug logs in the lookback functions for better debugging experience

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Base/ReleaseNotes/1_31_86.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Scripts
+
+##### CommonServerPython
+
+- Added debug logs for the look-back feature.
```
Comment (dantavori) [inline Packs/Base/ReleaseNotes/1_31_86.md:6]:
```suggestion
- Added debug logs for the look-back generic methods.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -10470,13 +10470,17 @@ def filter_incidents_by_duplicates_and_limit(incidents_res, last_run, fetch_limi
     :return: List of incidents after filtering duplicates when len(incidents) <= limit
     :rtype: ``list``
     """
+    demisto.debug('filtering incidents by duplicates and limit')
     found_incidents = last_run.get('found_incident_ids', {})
 
     incidents = []
+    
+    demisto.debug(f'number of incidents before filtering: {len(incidents_res)}')
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10478]: don't use fstring in CSP (it should be compatible with python 2) - relevant for all the next changes as well

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -10500,6 +10504,7 @@ def get_latest_incident_created_time(incidents, created_time_field, date_format=
     :return: The latest incident time
     :rtype: ``str``
     """
+    demisto.debug('Getting latest incident created time')
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10507]: let's print this after calculating it with its value

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -10529,6 +10534,7 @@ def remove_old_incidents_ids(found_incidents_ids, current_time, look_back):
     :return: The new incidents ids
     :rtype: ``dict``
     """
+    demisto.debug('Remove old incidents ids')
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10537]: where does the removal occur in this method?
also I wouldn't print this unless it contains a variable value

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -10608,14 +10616,16 @@ def create_updated_last_run_object(last_run, incidents, fetch_limit, look_back,
     :return: The new LastRun object
     :rtype: ``Dict``
     """
-
+    demisto.debug("Create updated last run object")
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10619]:
```suggestion
    demisto.debug("Create updated last run object, len(incidents) is {}, look_back is {}".format(len(incidents), look_back))
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 6]
```diff
@@ -10608,14 +10616,16 @@ def create_updated_last_run_object(last_run, incidents, fetch_limit, look_back,
...
     remove_incident_ids = True
 
     if len(incidents) == 0:
+        demisto.debug('len(incidents) == 0')
         new_last_run = {
             'time': end_fetch_time,
         }
     elif len(incidents) < fetch_limit or look_back == 0:
+        demisto.debug('len(incidents) < fetch_limit or look_back == 0')
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10628]:
```suggestion
        new_last_run = {
            'time': end_fetch_time,
        }
    elif len(incidents) < fetch_limit or look_back == 0:
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 7]
```diff
@@ -10470,13 +10470,17 @@ def filter_incidents_by_duplicates_and_limit(incidents_res, last_run, fetch_limi
     :return: List of incidents after filtering duplicates when len(incidents) <= limit
     :rtype: ``list``
     """
+    demisto.debug('filtering incidents by duplicates and limit')
     found_incidents = last_run.get('found_incident_ids', {})
 
     incidents = []
+    
+    demisto.debug(f'number of incidents before filtering: {len(incidents_res)}')
     for incident in incidents_res:
```
Comment (yaakovpraisler) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10479]: Let's print the incidents list before and after the filtering (or maybe just their ids with the given `id_field`)

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 8]
```diff
@@ -10538,6 +10544,7 @@ def remove_old_incidents_ids(found_incidents_ids, current_time, look_back):
         if current_time - addition_time < deletion_threshold_in_seconds:
             new_found_incidents_ids[inc_id] = addition_time
 
+    demisto.debug(f'Number of new found ids: {len(new_found_incidents_ids)}')
```
Comment (yaakovpraisler) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:10547]: Let's print the ids before and after the remove

**Review by yaakovpraisler — APPROVED** [review]

**Pull request merged**